src/cargo_loading_service.cpp

```diff
@@ -12,150 +12,178 @@
 
   // Publisher
-  pub_cargo_loading_state_ = this->create_publisher<v2i_interface_msgs::msg::InfrastructureCommandArray>(
-    "/cargo_loading/infrastructure_commands", rclcpp::QoS{3}.transient_local());
+  pub_commands_ = this->create_publisher<InfrastructureCommandArray>(
+    "/cargo_loading/infrastructure_commands", 10);
 
   // Subscriber
-  sub_parking_state_ =
-    this->create_subscription<in_parking_msgs::msg::InParkingStatus>(
+  sub_inparking_status_ = this->create_subscription<InParkingStatus>(
     "/in_parking/state", rclcpp::QoS{1},
-    std::bind(&CargoLoadingService::onParkingState, this, std::placeholders::_1),
-    subscribe_option);
-  sub_cargo_loading_state_ =
-    this->create_subscription<v2i_interface_msgs::msg::InfrastructureStateArray>(
+    std::bind(&CargoLoadingService::onInParkingStatus, this, _1), subscribe_option);
+  sub_infrastructure_status_ = this->create_subscription<InfrastructureStateArray>(
     "/infrastructure_status", rclcpp::QoS{1},
-    std::bind(&CargoLoadingService::onCargoLoadingState, this, std::placeholders::_1),
-    subscribe_option);
+    std::bind(&CargoLoadingService::onInfrastructureStatus, this, _1), subscribe_option);
+
+  // timer
+  const auto period_ns = rclcpp::Rate(command_pub_hz_).period();
+  timer_ = create_timer(
+    this, get_clock(), period_ns, std::bind(&CargoLoadingService::onTimer, this),
+    callback_group_subscription_);
+
+  // サービスcall時にtimerが回るように、最初にキャンセルしておく
+  timer_->cancel();
 }
 
 void CargoLoadingService::execCargoLoading(
-  const in_parking_msgs::srv::ExecuteInParkingTask::Request::SharedPtr request,
-  const in_parking_msgs::srv::ExecuteInParkingTask::Response::SharedPtr response)
+  const ExecuteInParkingTask::Request::SharedPtr request,
+  const ExecuteInParkingTask::Response::SharedPtr response)
 {
-  using InfrastructureCommand = v2i_interface_msgs::msg::InfrastructureCommand;
-  using InfrastructureCommandArray = v2i_interface_msgs::msg::InfrastructureCommandArray;
-  using InParkingStatus = in_parking_msgs::msg::InParkingStatus;
-  using ExecuteInParkingTaskResponse = in_parking_msgs::srv::ExecuteInParkingTask::Response;
+  // 設備ID取得
+  facility_id_ = request->value;
+
+  // aw_stateのsubscribe確認、NONEならばreject
+  if (aw_state_ == InParkingStatus::NONE) {
+    RCLCPP_WARN(this->get_logger(), "aw_state is NONE, reject...");
+    response->state = ExecuteInParkingTask::Response::FAIL;
+    return;
+  }
 
-  uint8_t response_state = ExecuteInParkingTaskResponse::SUCCESS;
-  bool cmd_pub = true;
-  int32_t aw_state;
+  // 設備連携要求開始
+  if (timer_->is_canceled()) {
+    timer_->reset();
+    RCLCPP_INFO(this->get_logger(), "Timer restart");
+  }
 
-  // Create command
-  InfrastructureCommand command;
-  command.type = "eva_beacon_system";
-  command.id = request->value;
+  // キャンセルになるまで設備連携要求を投げ続ける
+  while (!timer_->is_canceled()) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 1000 /* ms */, "request is running");
+  }
```

Comment:
```suggestion
  // キャンセルになるまで設備連携要求を投げ続ける
  while (1) {
    rclcpp::sleep_for(rclcpp::Rate(command_pub_hz_).period());
    RCLCPP_INFO_THROTTLE(
      this->get_logger(), *this->get_clock(), 1000 /* ms */, "request is running");
    std::lock_guard<std::mutex> lock(mutex_);
    if (timer_->is_canceled()) {
      break;
    }
  }
```
- sleepしないとCPU負荷がやばいので追加をお願いします。
- timer_の排他制御に対策が必要と思います。

Reply:
たしかにsleepを入れたほうが良さそう
値を書き換えるわけではなければmutexを入れる必要はないです

Reply:
sleepは、対応済み。mutexは不要となります。